lib/still/data.ex

```diff
@@ -0,0 +1,162 @@
+defmodule Still.Data do
+  import Still.Utils
+
+  use GenServer
+
+  @supported_extensions [".yml", ".exs", ".ex", ".json"]
+
+  @initial_state %{global: %{}}
+
+  @default_folder "_data"
+
+  @moduledoc """
+  Loads data files and makes data available in templates.
+  Any file in the _#{@default_folder}_ folder will be loaded
+  using the file's name as a key. For instance, a file in
+  _#{@default_folder}/site.json_ with the contents:
+
+      {
+        "title": "Still"
+      }
+
+  Will be available in the templates as `@site.title`.
+
+  You can also use folders to organise files; the same file in
+  `_#{@default_folder}/default/site.json_ would be available
+  in the templates as `@default.site.title`.
+
+  The data folder can be changed in the config:
+
+      config :still, Still.Data,
+        folder: "#{@default_folder}"
+
+  **Notice**: The data folder name should start with an underscore, otherwise
+  Still will consider the pages inside as web pages or assets to build.
+
+  It supports JSON (.json), YAML (.yml) and Elixir (.exs, .ex).
```

Comment:
Why these data types? Why not something like TOML as well? Not saying, I favour it, just trying to understand your choices

Reply:
I guess we can support all of them. I added these ones because they are very common as a data sources (except for EX but that's a must :p )